### DISCUSSION TITLE
QUIC with NextDNS only connecting to Ipv4-upstream?

### DISCUSSION THREAD

**Discussion by donald2612** [AdguardTeam/AdGuardHome#5289] [General]
Thank you for adopting very early to the new standard!

QUIC is working flawless with ipv4-upstream servers, but somehow no ipv6 upstreams are contacted. 
This isn't dramatic as Ipv6-addresses are resolved also on Ipv4 servers, but I wanted to know the background.
NextDNS support confirmed that the nodes were up and running.

Could it be that this is not implemented?

**Comment by ammnt** [comment]
> Thank you for adopting very early to the new standard!
> 
> QUIC is working flawless with ipv4-upstream servers, but somehow no ipv6 upstreams are contacted. This isn't dramatic as Ipv6-addresses are resolved also on Ipv4 servers, but I wanted to know the background. NextDNS support confirmed that the nodes were up and running.
> 
> Could it be that this is not implemented?

If you want to use their encrypted endpoints for DoH3 or DoQ connections and you want to be sure that connection will be going via IPv6, so try to encode it with SDNS stamps:
https://dnscrypt.info/stamps/

Specify IPv6 address and other settings and use it in AdGuard Home UI as upstream server🙂

**Comment by donald2612** [comment]
Yes, you could do this... or I can just resolve the correct IPv6 servers and use them directly, but that fully destroys the idea of DNS-failovers.
